interfaces/datastructure/StorageTrackable.h

```diff
@@ -0,0 +1,197 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef STORAGETRACKABLE_H
+#define STORAGETRACKABLE_H
+
+#include <nlohmann/json.hpp>
+
+#include "core/Log.h"
+#include "core/SerializationDefinitions.h"
+#include "core/SolARFrameworkDefinitions.h"
+#include "datastructure/MathDefinitions.h"
+#include "datastructure/StorageWorldElement.h"
+#include "datastructure/Trackable.h"
+#include "datastructure/UnitSystem.h"
+
+namespace SolAR {
+namespace datastructure {
+
+
+
+///
+/// @brief Enumeration of all trackable object types according to the API specified in partnership with the Etsi
+///
+enum class StorageTrackableType : char {
+    FIDUCIAL_MARKER,
+    IMAGE_MARKER,
+    MESH,
+    MAP,
+    GEOPOSE,
+    OTHER
+};
+
+static StorageTrackableType resolveTrackableType(std::string input){
+    if(input == "MAP"){return StorageTrackableType::MAP;}
+    if(input == "MESH"){return StorageTrackableType::MESH;}
+    if(input == "FIDUCIAL_MARKER" || input == "FIDUCIAL"){return StorageTrackableType::FIDUCIAL_MARKER;}
+    if(input == "IMAGE_MARKER" || input == "IMAGE"){return StorageTrackableType::IMAGE_MARKER;}
+    if(input == "GEOPOSE"){return StorageTrackableType::GEOPOSE;}
+    return StorageTrackableType::OTHER;
+}
+
+static  std::string resolveTrackableType(StorageTrackableType input){
+    if(input == StorageTrackableType::MAP){return "MAP";}
+    if(input == StorageTrackableType::MESH){return "MESH";}
+    if(input == StorageTrackableType::FIDUCIAL_MARKER){return "FIDUCIAL_MARKER";}
+    if(input == StorageTrackableType::IMAGE_MARKER){return "IMAGE_MARKER";}
+    if(input == StorageTrackableType::GEOPOSE){return "GEOPOSE";}
+    return "OTHER";
+}
+
+///
+/// @brief The EncodingInfo struct
+///
+class SOLARFRAMEWORK_API EncodingInfo {
+
+    public:
+
+        EncodingInfo() = default;
+
+        EncodingInfo(const std::string &dataFormat, const std::string &version){
+            m_dataFormat = dataFormat;
+            m_version = version;
+        }
+
+        std::string getVersion() const {
+            return m_version;
+        }
+
+        std::string getDataFormat() const {
+            return m_dataFormat;
+        }
+
+        void setDataFormat(const std::string &dataFormat){
+            m_dataFormat = dataFormat;
+        }
+
+        void setVersion(const std::string &version){
+            m_version = version;
+        }
+
+        friend class boost::serialization::access;
+        template<typename Archive>
+        void serialize(Archive &ar, const unsigned int version);
+
+    private:
+
+        std::string m_dataFormat;
+        std::string m_version;
+};
+
+/**
+    * @class StorageTrackable
+    * @brief <B>This class defines the generic trackable datastructure defined in the world graph.</B>
+    */
+class SOLARFRAMEWORK_API StorageTrackable : virtual public StorageWorldElement
```

Comment:
I'm not sure about the use of the word `Storage` everywhere. CMIIW, but I feel that what was initially refered to as WorldGraph and WorldStorage became shorten as Storage for convenience. But now, for someone new to the code, I'm not sure it really clear that all these types are called Storage.

Is StorageTrackable the class intended to replace Trackable, and was it named like this to avoid conflicts?

If so maybe we should think of new name and use namespaces maybe, for example:
```cpp
SolAR::datastructure::worlgraph::Trackable
SolAR::datastructure::worlgraph::Anchor
...
```
Or
```cpp
SolAR::datastructure::WorldGraphTrackable
SolAR::datastructure:::WorldGraphAnchor
```
Or it's considered a mouthful, even
```cpp
SolAR::datastructure::WGTrackable
SolAR::datastructure:::WGAnchor
```
What do you think?

Reply:
Done